horace_core/sqw/@sqw/parse_pixel_indices.m

```diff
@@ -1,212 +1,206 @@
+        % Assume ipix is an integer array - but it cannot be empty
+        if isempty(ipix)
+            error('HORACE:parse_pixel_indices:invalid_argument', ...
+                'Optional pixel index argument ''ipix'' is empty')
         end
-
         if nw == 1
-            irun = irun{1};
-            idet = idet{1};
-            ien = ien{1};
+            min_ipix = min(ipix(:));
+            max_ipix = max(ipix(:));
+            ipix_cellarray = {ipix};
+        else
+            min_ipix = min(ipix(:)) * ones(size(win));
+            max_ipix = max(ipix(:)) * ones(size(win));
+            ipix_cellarray = repmat({ipix}, sz_win);    % make a cell for convenience later on
         end
-
     end
-
-    return
-end
-
-% Case when given index array(s)
-% ------------------------------
-% Check iw, if given, is positive and consistent with the number of sqw objects
-if exist('iw','var')
-    if numel(iw)~=numel(win)
-        error('HORACE:parse_pixel_indices:invalid_argument', 'Number of indices in index argument ''iw'' must equal the number of sqw objects');
-    elseif any(iw<1)
-        error('HORACE:parse_pixel_indices:invalid_argument', 'Indices in index argument ''iw'' must be greater or equal to zero');
+    
+    % Get number of pixels in each sqw object
+    npix = arrayfun(@(x)(x.npixels), win);
+    
+    % Check consistency
+    bad = (min_ipix(:) < 1) | (max_ipix(:) > npix(:));
+    if any(bad)
+        ind = find(bad, 1);     % index of first bad dataset
+        error('HORACE:parse_pixel_indices:invalid_argument', ['One or more pixel ',...
+            'indices are out of the valid range 1-%d for dataset %d'], npix(ind), ind)
     end
 end
 
-% Check indx
-if ~iscell(indx)
-    indx = {indx};
-end
-
-nind = numel(indx);
-
-if any(~cellfun(@(x) isnumeric(x) && (size(x, 2) == 1 || size(x, 2) == 3), indx))
-    error('HORACE:parse_pixel_indices:invalid_argument', 'Each indexing array must be a numeric column vector or an nx3 array of run-detector-energy bin indices');
-end
-
-indx_internal = indx;        % standard name for internal working
-
-if nind==1
-    iw_internal = ones(nw,1);    % every sqw object has the same indx array - indx is effectively infinitely long
-else
-    if exist('iw','var')
-        if any(iw) > nind
-            error('HORACE:parse_pixel_indices:invalid_argument', 'Value(s) in the index argument ''iw'' must lie in the range 1 - %d', nind);
-        end
-        iw_internal = iw;
-    else
-        if nind ~= nw
-            error('HORACE:parse_pixel_indices:invalid_argument', 'If there is more than one indexing array, the number must match the number of sqw objects');
-        end
-        iw_internal = 1:nw;  % one indx array per sqw object
-    end
+% If no return arguments, simply have a silent return.
+% This goes against the default Matlab behaviour for functions, which is to
+% return the first output argument in a variable called 'ans'. However, that
+% could be fairly CPUtime expensive here.
```

Comment:
Is that the expectation of functions? That is just the role of the interpreter to handle cases where arguments are returned. I think this comment can be removed provided it is documented (and it is) in the docstrings.

Reply:
I added it in case someone queried the silent return. Happy to remove; done.